test/test_datapipe.py

```diff
@@ -377,6 +377,72 @@ def fn(item, dtype=torch.float, *, sum=False):
         for x, y in zip(map_dp_nl, input_dp_nl):
             self.assertEqual(x, torch.tensor(y, dtype=torch.float))
 
+    def test_map_datapipe_nested_level(self):
+
+        input_dp = IDP([list(range(10)) for _ in range(3)])
+
+        def fn(item, dtype=torch.float, *, sum=False):
+            data = torch.tensor(item, dtype=dtype)
+            return data if not sum else data.sum()
+
+        map_dp = input_dp.map(lambda ls: ls * 2, nesting_level=0)
```

Comment:
So, if I understand correctly (based on the naming of the `ls` variable), the `* 2` here operating on a list, so it's not that you're multiplying the integers by two, you are doubling the size of the list. Is this intentional?

Reply:
Yes, this is intentional.